apps/wpcom-block-editor/src/wpcom/features/tracking/wpcom-inserter-menu-search-term.js

```diff
@@ -42,11 +43,33 @@ const InserterMenuTrackingEvent = function () {
 		tracksRecordEvent( 'wpcom_block_picker_no_results', eventProperties );
 	}, 500 );
 
+	const debouncedSetFilterValueDiscovery = debounce( ( search_term ) => {
+		if ( search_term.length < 3 ) {
+			return;
+		}
+
+		const blockDirectoryResults = document.querySelectorAll(
```

Comment:
It would be good to have someone from the block editor review if this is the best way to check the results.

Reply:
Not a block editor expert, but as far as I can tell there's no equivalent filter for the block discovery results. You could create a PR to create something like `__unstableInserterMenuExtension` but for the block discovery results. In fact, that filter returns [a number of other things on top of `filterValue` and `hasItems`](https://github.com/WordPress/gutenberg/blob/ff13a7e4572c19eca9747f7f044386099cc60cb8/packages/block-editor/src/components/inserter/search-results.js#L183), so maybe it should return something about block discovery results 🤔 

But in the meantime I think we're stuck with a query string.